usecases/objects/references_add.go

```diff
@@ -22,66 +23,127 @@ import (
 	"github.com/semi-technologies/weaviate/usecases/objects/validation"
 )
 
-// AddObjectReference Class Instance to the connected DB. If the class contains a network
+// AddObjectReference to an existing object. If the class contains a network
 // ref, it has a side-effect on the schema: The schema will be updated to
 // include this particular network ref class.
-func (m *Manager) AddObjectReference(ctx context.Context, principal *models.Principal,
-	id strfmt.UUID, propertyName string, property *models.SingleRef) error {
-	err := m.authorizer.Authorize(principal, "update", fmt.Sprintf("objects/%s", id.String()))
-	if err != nil {
-		return err
+func (m *Manager) AddObjectReference(
+	ctx context.Context,
+	principal *models.Principal,
+	input *AddReferenceInput,
+) *Error {
+	deprecatedEndpoint := input.Class == ""
+	if deprecatedEndpoint { // for backward compatibility only
+		objectRes, err := m.getObjectFromRepo(ctx, "", input.ID, additional.Properties{})
+		if err != nil {
+			errnf := ErrNotFound{} // treated as StatusBadRequest for backward comp
+			if errors.As(err, &errnf) {
+				return &Error{"source object deprecated", StatusBadRequest, err}
+			}
+			return &Error{"source object deprecated", StatusInternalServerError, err}
+		}
+		input.Class = objectRes.Object().Class
+	}
+	path := fmt.Sprintf("objects/%s/%s", input.Class, input.ID)
+	if err := m.authorizer.Authorize(principal, "update", path); err != nil {
```

Comment:
I think that `authorizer.Authorize` method should be also run for case `deprecatedEndpoint == true`. Now we can run the code inside `if deprecatedEndpoint { ... }` even if we are not authorized to do that

Reply:
Just look closely authorizer.Authorize() doesn't depend on deprecatedEndpoint == true

Reply:
OK, my bad, I see it now, thanks!